cmd/mailbox-controller/controller.go

```diff
@@ -255,76 +258,76 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 func (ctl *mbCtl) ensureBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
 	logger := klog.FromContext(ctx).WithValues("mbsName", workspace.Name)
 
-	// The script ensures that a mailbox workspace has the needed KubeStellar CRDs.
 	// The script must be idempotent.
-	shellScriptName := "kubectl-kubestellar-ensure-mbw"
+	shellScriptName := "kube-bind"
 
-	logger.V(2).Info("Executing shell script", "script", shellScriptName)
-	// Make a sandbox for the concurrent access of the kubeconfig from the mailbox-controller
-	cmdLine := "cp $KUBECONFIG $KUBECONFIG.copy && KUBECONFIG=$KUBECONFIG.copy " + shellScriptName + " " + workspace.Name
-	logger.V(2).Info(cmdLine)
-	cmd := exec.Command("/bin/sh", "-c", cmdLine)
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stdout")
-		return true
-	}
-	stderr, err := cmd.StderrPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stderr")
-		return true
-	}
-	err = cmd.Start()
-	if err != nil {
-		logger.Error(err, "Unable to start executing the script")
-		return true
-	}
-	outBuf := bufio.NewReader(stdout)
-	errBuf := bufio.NewReader(stderr)
+	resourcesToBind := []string{"syncerconfigs", "edgesyncconfigs"}
+	for idx, resource := range resourcesToBind {
+		logger.V(2).Info("Ensuring binding", "script", shellScriptName, "resource", resource)
 
-	r, _ := regexp.Compile("🔒 (Created|Updated) secret.*namespace (.*)")
-	found, namespaceName := false, ""
-	for {
-		line, _, err := errBuf.ReadLine()
+		// Make sandboxes for the concurrent access of the kubeconfig from the mailbox-controller's multiple workers
+		// This is a compromise due to the situation that 1) we are using kcp and 2) we are running tests on bare process of this controller
+		workerIdx, _ := ctx.Value(ctxKey("worker")).(int)
```

Comment:
Discarding the error is dangerous. `workerIdx := ctx.Value(ctxKey("worker")).(int)` is valid and panics if the type assertion fails --- which is safer. Safer still would be to generate a fresh number every time (e.g., with `atomic.AddUint64(.., 1)`) and include deletion of the temp file in the command line (e.g., with a command like `( cp $src $temp; trap "rm -f $temp" EXIT; use $temp )`).

Reply:
Cool, good to know something new today and happy to see that works as added [here](https://github.com/kubestellar/kubestellar/pull/1301/commits/2d4bfd6c1d4d069ff4c8eec1a5eaed39fbe68af0).